azkaban-web-server/src/main/java/azkaban/webapp/AzkabanWebServer.java

```diff
@@ -142,53 +134,50 @@
   private final ExecutorManager executorManager;
   private final ScheduleManager scheduleManager;
   private final TriggerManager triggerManager;
-  private final ClassLoader baseClassLoader;
   private final MetricRegistry registry;
   private final MetricsManager metricsManager;
   private final Props props;
   private final SessionCache sessionCache;
   private final List<ObjectName> registeredMBeans = new ArrayList<>();
+
   //queuedThreadPool is mainly used to monitor jetty threadpool.
   private QueuedThreadPool queuedThreadPool;
   private Map<String, TriggerPlugin> triggerPlugins;
   private MBeanServer mbeanServer;
 
-  /**
-   * Constructor usually called by tomcat AzkabanServletContext to create the
-   * initial server
-   */
-  public AzkabanWebServer() throws Exception {
-    this(null, loadConfigurationFromAzkabanHome());
-  }
-
   @Inject
-  public AzkabanWebServer(final Server server, final Props props) throws Exception {
+  public AzkabanWebServer(final Props props,
+      final Server server,
+      final ExecutorManager executorManager,
+      final ProjectManager projectManager,
+      final TriggerManager triggerManager,
+      final MetricsManager metricsManager,
+      final MetricRegistry metricRegistry,
+      final SessionCache sessionCache,
+      final UserManager userManager,
+      final ScheduleManager scheduleManager,
+      final VelocityEngine velocityEngine) {
     this.props = requireNonNull(props);
-    this.server = server;
-
-    this.velocityEngine = configureVelocityEngine(props.getBoolean(VELOCITY_DEV_MODE_PARAM, false));
-    this.sessionCache = new SessionCache(props);
-    this.userManager = loadUserManager(props);
-
-    // TODO remove hack. Move injection to constructor
-    this.executorManager = SERVICE_PROVIDER.getInstance(ExecutorManager.class);
-    this.projectManager = SERVICE_PROVIDER.getInstance(ProjectManager.class);
-    this.triggerManager = SERVICE_PROVIDER.getInstance(TriggerManager.class);
-    this.metricsManager = SERVICE_PROVIDER.getInstance(MetricsManager.class);
-    this.registry = SERVICE_PROVIDER.getInstance(MetricRegistry.class);
+    this.server = requireNonNull(server);
+    this.executorManager = requireNonNull(executorManager);
+    this.projectManager = requireNonNull(projectManager);
+    this.triggerManager = requireNonNull(triggerManager);
+    this.metricsManager = requireNonNull(metricsManager);
+    this.registry = requireNonNull(metricRegistry);
+    this.sessionCache = requireNonNull(sessionCache);
+    this.userManager = requireNonNull(userManager);
+    this.scheduleManager = requireNonNull(scheduleManager);
+    this.velocityEngine = requireNonNull(velocityEngine);
```

Comment:
Nope. This isn't about Guice. It's just defensive programming. 
It is good to check arguments to make sure that you have lesser surprises.